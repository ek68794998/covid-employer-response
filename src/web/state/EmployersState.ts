import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}