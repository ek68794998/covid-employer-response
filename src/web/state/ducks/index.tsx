import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});