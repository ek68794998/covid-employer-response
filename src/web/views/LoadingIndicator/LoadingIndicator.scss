.LoadingIndicator__Container {
	font: 400 20px 'Baloo Chettan 2';
	margin: 16px auto;
	text-align: center;
}
