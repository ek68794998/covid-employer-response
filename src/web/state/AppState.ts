import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}