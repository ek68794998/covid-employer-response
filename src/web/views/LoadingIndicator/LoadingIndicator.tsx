import React from "react";
import { useSelector } from "react-redux";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import "./LoadingIndicator.scss";

const LoadingIndicator: React.FC = (): React.ReactElement => {
	const strings: LocalizedStrings = useSelector(getStrings);

	return (
		<div className="LoadingIndicator__Container">
			{strings.loading}
		</div>
	);
};

export default LoadingIndicator;
