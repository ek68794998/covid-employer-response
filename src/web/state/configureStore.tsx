import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;