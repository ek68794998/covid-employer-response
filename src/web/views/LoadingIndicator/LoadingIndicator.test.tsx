import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import LoadingIndicator from "./LoadingIndicator";

describe("<LoadingIndicator />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<LoadingIndicator />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
