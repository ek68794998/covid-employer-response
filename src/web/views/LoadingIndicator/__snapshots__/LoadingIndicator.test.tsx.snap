// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> renders without exploding 1`] = `
<div
  className="LoadingIndicator__Container"
/>
`;
